app/classifier/drawing-tools/polygon.cjsx

```diff
@@ -145,3 +169,24 @@ module.exports = createReactClass
     @props.mark.points[index].x += difference.x
     @props.mark.points[index].y += difference.y
     @props.onChange @props.mark
+
+  handleGhostHandleStart: (index, e, d) ->
+    if index == 0
+      startPoint = @props.mark.points[@props.mark.points.length]
+    else
+      startPoint = @props.mark.points[index - 1]
+    if index == @props.mark.points.length
+      endPoint = @props.mark.points[0]
+    else
+      endPoint = @props.mark.points[index]
+    
+    new_point = {
+      x: (endPoint.x + startPoint.x) / 2,
+      y: (endPoint.y + startPoint.y) / 2,
+    }
+    @props.mark.points = [
```

Comment:
the `mark` prop is being mutated - is this being done elsewhere? if so i'm just going to trust that it works as expected, but it feels like `onChange` should be called with a new `mark`?

Reply:
`newMark = {...@props.mark}`

but if this is how existing code is working let's stick with it. deviating from the existing approach is more likely to cause issues than doing it properly, in this case

Reply:
let's just not, the existing code is our guide here